provisioning/service/src/ProvisioningServiceClient.cs

```diff
@@ -121,12 +197,43 @@ private ProvisioningServiceClient(string connectionString, HttpTransportSettings
             /* SRS_PROVISIONING_SERVICE_CLIENT_21_003: [The constructor shall throw ArgumentException if the ProvisioningConnectionString or one of the inner Managers failed to create a new instance.] */
             /* SRS_PROVISIONING_SERVICE_CLIENT_21_004: [The constructor shall create a new instance of the ContractApiHttp class using the provided connectionString.] */
             _provisioningConnectionString = ServiceConnectionString.Parse(connectionString);
+            _hostName = _provisioningConnectionString.HostName;
             _contractApiHttp = new ContractApiHttp(
                 _provisioningConnectionString.HttpsEndpoint,
                 _provisioningConnectionString,
                 httpTransportSettings);
         }
 
+        private ProvisioningServiceClient(string hostName, TokenCredential credential, HttpTransportSettings httpTransportSettings)
+        {
+            if (string.IsNullOrWhiteSpace(hostName ?? throw new ArgumentNullException(nameof(hostName))))
+            {
+                throw new ArgumentException($"{nameof(hostName)} cannot be empty string");
```

Comment:
`ArgumentException` has a ctor combo that takes the name of the parameter independent of the message. It is preferred to use.